capstone/src/test/java/com/google/step/servlets/LoginServletTest.java

```diff
@@ -0,0 +1,125 @@
+        // user is not even logged in.
+    }
+
+    @Test
+    public void testDoGetWithoutLogIn() throws IOException, SAXException {
+        // Testing doGet method without having the user logged in.
+        // Result: {"LoggedIn":"false","loginURL":"someURL"}
+
+        ServletRunner sr = new ServletRunner();
+        // The first parameter has to be the same as whatever goes after the '/' in the urlString for
+        // the GetMethodWebRequest so for example if the first parameter is 'restaurant' then the link
+        // would be: 'http://localhost:8080/restaurant' even though the actual servlet may be called
+        // something else.
+        sr.registerServlet("login", LoginServlet.class.getName());
+        ServletUnitClient sc = sr.newClient();
+
+        // This can throw the SAXException.
+        WebRequest request = new GetMethodWebRequest("http://localhost:8080/login");
+        WebResponse response = sc.getResponse(request);
+        assertNotNull("No response received", response);
+        assertEquals("content type", "application/json", response.getContentType());
+
+        // Turn content to JSon
+        JsonObject jsonObject = turnStringToJson(response.getText());
+
+        assertNotNull("Return Json", response.getText());
+        assertEquals("Json Loggedin", "false", jsonObject.get("LoggedIn").toString());
+
+        UserService userService = UserServiceFactory.getUserService();
+        assertEquals("Login URL", "\"" + userService.createLoginURL("/") + "\"", jsonObject.get("loginURL".toString()).toString());
+    }
+
+    @Test
+    public void testDoGetWithLogIn() throws IOException, SAXException {
+        // If you set an email you must also set the auth domain.
+        helper.setEnvIsLoggedIn(true)
+                .setEnvIsAdmin(false)
+                .setEnvEmail("example@gmail.com")
+                .setEnvAuthDomain("example@gmail.com");
+
+        ServletRunner sr = new ServletRunner();
+
+        sr.registerServlet("login", LoginServlet.class.getName());
+        ServletUnitClient sc = sr.newClient();
+
+        WebRequest request = new GetMethodWebRequest("http://localhost:8080/login");
+        WebResponse response = sc.getResponse(request);
+
+        assertNotNull("No response received", response);
+        assertEquals("content type", "application/json", response.getContentType());
+
+        JsonObject jsonObject = turnStringToJson(response.getText());
+
+        assertEquals("LoggedIn", "true", jsonObject.get("LoggedIn").toString());
+
+        // Have to add the extra quotes in the expected because for some reason the JSON Parser adds
+        // them again.
+        assertEquals("Email", "\"example@gmail.com\"", jsonObject.get("Email").toString());
+
+        UserService userService = UserServiceFactory.getUserService();
+        assertEquals("Log out URL", "\"" + userService.createLogoutURL("/") + "\"", jsonObject.get("logOutURL").toString());
```

Comment:
Hmmm how come there's quotes embedded into your logout url? Is it because that's the redirect URL and is passed as a parameter to the logout url?

Reply:
For some reason, when I parse the json String in the turnStringToJson() method, it adds an extra set of quotes as part of the string itself. For now this is the quick solution I found.